### DISCUSSION TITLE
Archived server downloads faster?

### DISCUSSION THREAD

**Discussion by engvalla** [wildfly/wildfly-maven-plugin#211] [General]
I've been testing the 3.0.0 alphas and they seems slower then the 2.x.x versions. We don't use that much containers so a full server that is the same as the one that operations downloads and installs seems like a perfectly valid use case for us. And even where we have started to use containers, a full server in a docker layer and a thin war in a separate layer seems faster then rebuilding the entire server every time. Is there an option to keep the archived server downloads, yes we can stick with the 2.x.x version I know, but I thought I'd ask at least

regards
Andreas

**Comment by jamezp** [comment]
@engvalla The support for downloading the full archive was removed in 3.0. I would imagine it's only faster when the download already exists in the local repository. However, this is valuable input.

Do you generally use the full server? I'm only asking because there is a way to speed it up by not configuring a full server. @jfdenise do you have any input?

**Reply by engvalla** [reply]
We use the full server and standalone-full-ha.xml on 1000+ servers, all configured basically in the same way. Local cached maven repositories and cached docker layers where we use containers is helping a lot with the speed yes.

Any speed improvements are of interest, yes.

**Reply by jamezp** [reply]
Could you give a sample of what the plugin configuration looks like and maybe which goal you use? It might be helpful to understand your use-case as maybe there is a way to speed it up already.

**Reply by engvalla** [reply]
sure, in pom.xml we use

    <plugin>
        <groupId>org.wildfly.plugins</groupId>
        <artifactId>wildfly-maven-plugin</artifactId>
        <version>2.1.0.Final</version>
        <configuration>
            <addUser>
                <users>
                    <user>
                        <username>admin</username>
                        <password>admin</password>
                    </user>
                </users>
            </addUser>
            <propertiesFile>deployment/jboss/cli.properties</propertiesFile>
            <javaOpts>
                <javaOpt>-javaagent:${settings.localRepository}/org/jacoco/org.jacoco.agent/0.8.7/org.jacoco.agent-0.8.7-runtime.jar=destfile=target/jacoco-it.exec</javaOpt>
            </javaOpts>
            <serverConfig>standalone-full-ha.xml</serverConfig>
            <scripts>
                <script>../deployment/jboss/h2.cli</script>
                <script>../deployment/jboss/jms-queue.cli</script>
                <script>../deployment/jboss/system-properties.cli</script>
            </scripts>
            <version>23.0.2.Final</version>
        </configuration>
    </plugin>

and in the Jenkinsfile we use

    steps {
        sh "mvn wildfly:start"
        sh "mvn wildfly:execute-commands"
        sh "mvn wildfly:deploy-only"
        sh "mvn failsafe:integration-test failsafe:verify"
        sh "mvn wildfly:shutdown"
        sleep(time: 3, unit: "SECONDS") // jacoc-it.exec needs some time to be written down
        sh "mvn org.jacoco:jacoco-maven-plugin:0.8.7:report-integration"
    }

**Reply by jfdenise** [reply]
When using standalone-full-ha.xml, we need the full server. We don't have all galleon layers to cover this default configuration so we can't benefit from trimming.

**Reply by jamezp** [reply]
> When using standalone-full-ha.xml, we need the full server. We don't have all galleon layers to cover this default configuration so we can't benefit from trimming.

Yeah, I'm just wondering if we could figure out a way to somehow speed it up. When using the `org.wildfly:wildfly-dist:zip:$version` the full distribution is already download so it can quickly be copied and uncompressed. Using Galleon it takes time to fully assemble everything.

I'm not really suggesting we add back the option to use the dist, but it's more something to think about if there is a way to do make some performance improvements if we know we need everything.
